chardevice.go

```diff
@@ -170,10 +170,18 @@ func (cdev CharDevice) QemuParams(config *Config) []string {
 	}
 
 	// Legacy serial is special. It does not follow the device + driver model
-	if cdev.Driver != LegacySerial {
+	if cdev.Driver != LegacySerial && cdev.Driver != PCISerial {
 		qemuParams = append(qemuParams, "-device")
 		qemuParams = append(qemuParams, strings.Join(deviceParams, ","))
 	}
+	
+	//appending -serial none and -monitor none to qemuparams if we are using pci-serial
+	if cdev.Driver == PCISerial {
+		qemuParams = append(qemuParams, "-serial")
+		qemuParams = append(qemuParams, "none")
```

Comment:
I think this _may_ be required on qemu-system-aarch64, we can keep this for now;  Though I think we may not know whether an image we boot uses console=ttyS0 or console=ttyAMA0.  Need to think on this.

Reply:
OK, this isn't your fault -- with a few months off  from looking at this code I'm currently bothered by how in serialdevice_test.go, I have mostly Chardevices, and in chardev_test, all of them are using LegacySerial so it doesnt' emit a -device; but chardevs don't have devices at all, they have backends, which I've already modeled.

So, serialdevice and chardevice will need a refactor but let's merge this stuff first.

Reply:
Yeah I noticed that too. I can start drafting up changes for this after we get it merged